build-source-tarball.sh

```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 IFS=$'\n\t'
 
 usage() {
-    echo "usage: $0 <path-to-tarball-root> [--skip-build] [--enable-leak-detection] [--skip-prebuilt-check] [-- [extra build.sh args]]"
+    echo "usage: $0 <path-to-tarball-root> [--skip-build] [--enable-leak-detection] [--skip-prebuilt-check] [--custom-ref-packages <dir>] [--custom-previously-built-packages <dir>] [-- [extra build.sh args]]"
```

Comment:
If I am understand this right, this will work like this:

0. User builds `source-build-reference-packages`
1. User builds `source-build`
2. User calls `./build-source-tarball.sh --custom-ref-packages /path/to/source-build-reference-packages --custom-previously-built-packages /path/to/source-build`
3. The tarball generated contains copies of the older source-build from step 1 and source-build-reference-packges from step 0? The artifacts from step 0 and step 1 can be safely deleted after the tarball has been built?

Is that right?

Reply:
Oh, that would be a more convenient way to do this.  Currently they would be needed during `build.sh` out of the tarball as well, but that seems a bit more user-friendly unless you have a preference for keeping things out of the tarball.

Reply:
I have a very strong preference for reducing the tarball size :) A number of build environments dont work very well when we start throwing 2GB+ sources at them. 

Plus, it makes the tarball less portable if it contains prebuilt bits. It's also hard to audit which of those prebuilts are from an older "source-build" vs which ones are prebuilts accidentally inserted into the sources.

I was going to ask for the opposite: would it be possible to not bundle anything into the tarball and requires that users pass in the path from step 0 and step 1 when building the contents of the tarball? Is that doable? Does it require step 0 and step 1 in that order? Or could we modify and rebuild `source-build` and supply an updated SDK (eg, bugfixes) to the build from the tarball?

Reply:
Yes, that's actually the direction I was heading.  We could actually do this even just requiring a list of packages in step 2 - if we set it up that way you could do something like
1. `./build-source-tarball.sh --ref-package-list ref-packages.txt --previously-built-package-list previously-built.txt`
2. From the generated tarball, `./build.sh --custom-ref-packages /path/to/source-build-reference-packages --custom-previously-built-packages /path/to/previously-built`

And swap out those two sets of packages however/whenever you want.  The thing that matters is that the tarball is checking that all prebuilt packages are either [known extras](https://github.com/dotnet/source-build/blob/1ae9cc75797a218449f6a66d32abb35549ee6be2/support/additional-prebuilts-to-delete.txt), known ref packages, or known available source-built packages.  Does that make sense?

Reply:
Yes, it does. Thanks for clarifying. Can these invocations be added to the usage/help messages of the scripts?

Reply:
Yes, I'll fix up the help.  Thanks!